### DISCUSSION TITLE
Custom themes are not getting applied to my Next Js project while Migrating Chakra UI from v2 to v3

### DISCUSSION THREAD

**Discussion by ganeshKolavennuBb** [chakra-ui/chakra-ui#9125] [Q&A] [answered]
Hey everyone I was trying to migrate Chakra from v2 to v3. Even though following everystep by the docs I am still not seeing my custom theme getting applied. Here are my tech-stack specifications.

FYI. I've tried everything and still couldn't get it loaded

### Tech Stack
- Next Js (v.14)
- Chakra UI (v.3)

### System Specification

- [x] Linux
- [ ] Windows
- [ ] Mac

### Here are my Theme Provider, Themes file

Theme Provider
```js
'use client'

import { ChakraProvider } from '@chakra-ui/react';
import { system } from './theme';

export function ChakraThemeProvider({ children }) {
  return <ChakraProvider value={system}>{children}</ChakraProvider>;
}
```

Theme File
```js
import {
  createSystem,
  defaultConfig,
  defineConfig,
  mergeConfigs,
  defaultBaseConfig,
} from '@chakra-ui/react';
import { textStyles } from '../../themes/textstyles';

const config = defineConfig({
  theme: {
    extend: {
      textStyles,
    },
  },
});

export const system = createSystem(defaultConfig, config);
```
TextStyles Typography

```js
import { defineTextStyles } from '@chakra-ui/react';

export const textStyles = defineTextStyles({
  body3: {
    value: {
      fontSize: '12px',
      lineHeight: '14px',
    },
  },
});

```
What am I doing wrong ? Is there any anomaly.

**Comment by vitalijalbu** [comment]
on mine too, I'd also want to have the theme code in a separated file... is that possible?

**Comment by segunadebayo** [comment, marked as answer]
`textStyles` do not restyle every `Text` by themselves. Pass the style:

```tsx
<Text textStyle="body3">...</Text>
```

`theme.extend` is fine. Confirm `ChakraProvider value={system}` wraps the tree that renders the text.
